crates/storage/provider/src/test_utils/mock.rs

```diff
@@ -42,7 +42,7 @@ use std::{
 
 /// A mock implementation for Provider interfaces.
 #[derive(Debug, Clone)]
-pub struct MockEthProvider {
+pub struct MockEthProvider<T = TransactionSigned> {
```

Comment:
this alone doesn't do much yet,

could we maybe relax the impls as well and use this tx type in `Blocks`?

perhaps a better generic would actually be `B: Block`